### ISSUE TITLE
shutter priority

### ISSUE THREAD

**Issue by petrica-celrau** [troop/FreeDcam#140]
Hi Ingo, this is a feature I really miss and couldn't find on any camera app I've tried, set the shutter speed manually and have ISO adjust automatically  for the proper exposure. So many times I got unsharp images due to long exposure times in auto mode and while the full manual mode is good to have, it's not the fastest way to take pictures.

**Comment by troop** [comment]
on newer qcom socs it should be possible. but never saw a working implementation.
with your device its not possible.
that it works with iso is caused due api1. it limits the ae iso to a exposure table.
the exposure table is then a lookup how fast/slow the shutter can get.
with camera2 its possible to create some kind of own ae handler. but by default it supports only full manual or auto

**Comment by petrica-celrau** [comment]
Ingo, my current device is Redmi Note 3 (SD) with camera2 enabled in
build.prop, just to clarify which is 'my device'. I gave up on that HTC one
mini2.
Regarding shutter priority (or ISO priority for that matter), I figured
that camera2 on its own only supports all auto or all manual. In auto mode
looks like there is a preference for low iso just to keep noise at minimum
which in cases with less then ideal lighting means longer exposure time, at
say 1/10s I would have a slim chance to get a sharp picture without a
tripod. The way I was thinking about shutter priority is quite simplistic,
you take auto exposure, say 125iso@1/12s and calculate iso for say, 1/125s
to give you the same exposure; if the resulting iso is higher then
available, user should be able to choose between a lower speed or accept
underexposure. I'm guessing this is what you mean by 'own ae handler'.
How possible could that be?
Can't help thinking like the photographer that I am :)

Also, I found the latest builds quite unstable, the camera app stops very
often, let me know if you need logs. I'm on a 7.1.1 rom based on AOSP
<https://forum.xda-developers.com/redmi-note-3/development/rom-nitrogen-os-11-01-2017-t3536211>
but
I've also tested with CM13 and the app is not more stable.

**Comment by petrica-celrau** [comment]
Yeah, camera2. If you could implement some semi-manual controls that would be great, no camera app has that. Ideally, I'd like to have auto-focus, auto-iso, manual speed.
Also in settings we have Camera Control: off, I think 'manual' would be more appropriate then 'off'.

**Comment by troop** [comment]
some day it will happen, but i kicked freedcam into alpha state^^
that naming is based on the camera api. but i replace stuff one by one. 
shutter or iso priority has less priority for me. so dont expect it soon

**Comment by petrica-celrau** [comment]
Understood, thanks.
